core/geometry/package.json

```diff
@@ -39,28 +39,16 @@
   "devDependencies": {
     "@itwin/build-tools": "workspace:*",
     "@itwin/eslint-plugin": "^4.0.2",
-    "@types/chai": "4.3.1",
     "@types/flatbuffers": "~1.10.0",
-    "@types/mocha": "^10.0.6",
     "@types/node": "~18.16.20",
-    "chai": "^4.3.10",
-    "debug": "^2.6.9",
+    "@vitest/coverage-istanbul": "^2.1.0",
     "eslint": "^8.56.0",
-    "mocha": "^10.2.0",
-    "nyc": "^15.1.0",
     "rimraf": "^3.0.2",
-    "typescript": "~5.3.3"
+    "typescript": "~5.3.3",
+    "vitest": "^2.1.0"
   },
   "dependencies": {
     "@itwin/core-bentley": "workspace:*",
     "flatbuffers": "~1.12.0"
-  },
-  "nyc": {
```

Comment:
@aruniverse @saeeedtorabi did this work in the past? I copied the functionality of the coverage below into geometry's vitest.config.mts and it's now failing coverage, saying `Coverage for functions (53.06%) does not meet global threshold (85%)`

Reply:
Currently on the master branch our coverage for `core-geometry` is 
```
Statements   : 93.53% ( 38252/40895 )
Branches     : 82.55% ( 15344/18587 )
Functions    : 92.47% ( 6318/6832 )
Lines        : 93.53% ( 36824/39368 )
```

@dassaf4 Any idea why @hl662  is getting 53% coverage?!

Reply:
I'll check for regressions - maybe the one in master excludes certain file paths

Reply:
vitest's `expect` has different syntaxes than chai's `expect`, perhaps when we use the old syntax, that isn't caught by the coverage - I'll refactor all of the `expect` methods and see if it helps

Reply:
take a look at `tools/build/.nycrc`

Reply:
none on the test files - I'm looking at the local coverage report right now, there's no `test.ts`

Reply:
I see only one core-geometry test file with the `expect` calls changed to `expectTo`. I assume this is an experiment and the goal is to match the current coverage percentages with the new test framework. Otherwise, core-geometry changes look good so far. Let us know if you need help @hl662!

Reply:
![image](https://github.com/user-attachments/assets/a03a838c-9755-4159-b78c-6d31d83c06f1)
Update: I updated the vitest config coverage's `include` and `exclude` and got the numbers above, we might be missing some covered lines and statements by a small fraction, lmk what you think @saeeedtorabi @dassaf4

Reply:
Sweet! I'm fine with those numbers. These are estimates of estimates, so the decimal differences are unimportant. Besides, according to L. Kronecker, "God created the integers, all else is the work of man." ;)